service/src/main/java/org/kiwiproject/champagne/resource/TagResource.java

```diff
@@ -0,0 +1,105 @@
+package org.kiwiproject.champagne.resource;
+
+import static java.util.Objects.isNull;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.kiwiproject.champagne.model.AuditRecord.Action.DELETED;
+import static org.kiwiproject.champagne.util.DeployableSystems.checkUserAdminOfSystem;
+import static org.kiwiproject.champagne.util.DeployableSystems.getSystemIdOrThrowBadRequest;
+
+import javax.annotation.security.PermitAll;
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import com.codahale.metrics.annotation.ExceptionMetered;
+import com.codahale.metrics.annotation.Timed;
+import lombok.extern.slf4j.Slf4j;
+import org.kiwiproject.champagne.dao.AuditRecordDao;
+import org.kiwiproject.champagne.dao.TagDao;
+import org.kiwiproject.champagne.model.AuditRecord;
+import org.kiwiproject.champagne.model.Tag;
+import org.kiwiproject.dropwizard.error.dao.ApplicationErrorDao;
+
+@Path("/tag")
+@Produces(APPLICATION_JSON)
+@Consumes(APPLICATION_JSON)
+@PermitAll
+@Slf4j
+public class TagResource extends AuditableResource {
+
+    private final TagDao tagDao;
+
+    public TagResource(TagDao tagDao, AuditRecordDao auditRecordDao, ApplicationErrorDao errorDao) {
+        super(auditRecordDao, errorDao);
+
+        this.tagDao = tagDao;
+    }
+
+    @GET
+    @Timed
+    @ExceptionMetered
+    public Response listTagsForCurrentSystem() {
+        var systemId = getSystemIdOrThrowBadRequest();
+
+        var tags = tagDao.findTagsForSystem(systemId);
+
+        return Response.ok(tags).build();
+    }
+
+    @POST
+    @Timed
+    @ExceptionMetered
+    public Response createTag(@Valid Tag tag) {
+        checkUserAdminOfSystem();
+
+        if (isNull(tag.getDeployableSystemId())) {
+            var systemId = getSystemIdOrThrowBadRequest();
+            tag = tag.withDeployableSystemId(systemId);
+        }
+
+        var tagId = tagDao.insertTag(tag);
+
+        auditAction(tagId, Tag.class, AuditRecord.Action.CREATED);
```

Comment:
Are these audits executed sync or async? Might want to execute them async if not already. Then again, that might be complete overkill.

Reply:
They are sync, but all they do is insert in the db